### PR TITLE
Feature/coach check service

### PR DESCRIPTION
**Pull request by ExxZzyy**
Feature/coach check service

### CODE REVIEW

File: src/app/models/coach-profile/coach-profile.module.ts [hunk 1]
```diff
@@ -15,6 +16,7 @@ import { CoachProfileEditorModule } from '../coach-profile-editor/coach-profile-
     CoachProfileRoutingModule,
     CoachProfileDialog,
     CoachProfileEditorModule,
+    MatProgressSpinnerModule,
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile.module.ts:19]: do you need this?

File: src/app/models/coach-profile/coach-profile.component.ts [hunk 2]
```diff
@@ -1,27 +1,47 @@
...
     CoachTestTabs.checked,
   ];
 
-  tables: { [key: string]: TestData[] } = {};
+  tables: { [key: string]: CoachTest[] } = {};
 
-  constructor(public dialog: MatDialog) {
+  constructor(public dialog: MatDialog, private coachTestStoreService: CoachTestsStoreService) {
     this.tables = {
       [CoachTestTabs.unchecked]: MOCK_TESTS.filter((test) => !test.isChecked),
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile.component.ts:28]: why you still need mocks here?

File: src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <div class="mat-elevation-z8">
-  <table mat-table [dataSource]="dataSource" matSort matSortActive="date" matSortStart="asc">
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.html:2]: how you use dataSource in component then?

File: src/app/models/coach-profile/coach-profile.module.ts (hunk 1 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile.module.ts:19]: Yes, for this
image

File: src/app/models/coach-profile/coach-profile.component.ts (hunk 2 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile.component.ts:28]: done
f3c1edc927ccaac75f32224f68afffbadcb79e6b

File: src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.html (hunk 3 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.html]: done
f3c1edc927ccaac75f32224f68afffbadcb79e6b

File: src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts [hunk 4]
```diff
@@ -66,17 +74,12 @@ export class CoachProfileTableComponent implements AfterViewInit, OnChanges, OnI
...
-      this.dataSource.data = changes.table.currentValue;
-    }
-  }
-
   onClick(id: number) {
     this.dialog.open(CoachProfileDialogComponent, { data: { id } });
   }
 
   ngOnInit(): void {
+    // this.dataSource = new MatTableDataSource(this.results);
```
Comment (RomPavel) [inline src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts:82]: remove this

File: src/app/models/coach-profile/coach-profile.component.ts (hunk 2 above)
Comment (RomPavel) [reply src/app/models/coach-profile/coach-profile.component.ts:28]: I dont understand, you have mocks for **this.tables** and for **this.results$**, why you need both of them?

File: src/app/models/coach-profile/service/coach-tests-store.service.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      .pipe(concatMap(() => this.coachTestHttpService.getCheckedCoachTests()))
+      .subscribe({
+        next: (res) => {
+          this.coachTestResults = [...res];
+        },
+      });
+  }
+
+  getCoachUncheckedTestResults(): void {
+    this.authStoreService.activeUser$
```
Comment (RomPavel) [inline src/app/models/coach-profile/service/coach-tests-store.service.ts:45]: Why you need to wait for active user here?

File: src/app/models/coach-profile/service/coach-tests-store.service.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      .pipe(concatMap(() => this.coachTestHttpService.getHighPriorityCoachTests()))
+      .subscribe({
+        next: (res) => {
+          this.coachTestResults = [...res];
+        },
+      });
+  }
+
+  getCoachCheckedTestResults(): void {
+    this.authStoreService.activeUser$
```
Comment (RomPavel) [inline src/app/models/coach-profile/service/coach-tests-store.service.ts:35]: same here

File: src/app/models/coach-profile/service/coach-tests-store.service.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    this.coachTestSubject$.next(coachTestResults);
+  }
+
+  constructor(
+    private coachTestHttpService: GetCoachTestsHttpService,
+    private authStoreService: AuthStoreService,
+  ) {}
+
+  getCoachHighPriorityTestResults(): void {
+    this.authStoreService.activeUser$
```
Comment (RomPavel) [inline src/app/models/coach-profile/service/coach-tests-store.service.ts:25]: same here

File: src/app/models/coach-profile/service/get-coach-tests-http.service.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { BASE_API_URL } from '../../../constants/route-constant';
+
+export interface CoachTest {
+  id: string;
+  level: string;
+  isChecked: boolean;
+  isHigh: boolean;
+  date: Date;
```
Comment (RomPavel) [inline src/app/models/coach-profile/service/get-coach-tests-http.service.ts:12]: are you sure that backend return Date object to us?

File: src/app/models/coach-profile/coach-profile.module.ts (hunk 1 above)
Comment (RomPavel) [reply src/app/models/coach-profile/coach-profile.module.ts:19]: Do you really think we need 2nd spinner?
Yours spinner just spin infinitely, even when request is done, while 1st spinner only spin while making request 
image

Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile.module.ts]: Delete it b6aebad9c2c3ea2c4d560dcc67d65c872c630642

File: src/app/models/coach-profile/coach-profile.component.ts (hunk 2 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile.component.ts:28]: Remove mocks from results$ b6aebad9c2c3ea2c4d560dcc67d65c872c630642

File: src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts (hunk 4 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/coach-profile-table/coach-profile-table.component.ts]: Done b6aebad9c2c3ea2c4d560dcc67d65c872c630642

File: src/app/models/coach-profile/service/coach-tests-store.service.ts (hunk 5 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/service/coach-tests-store.service.ts]: Don't need, remove b6aebad9c2c3ea2c4d560dcc67d65c872c630642

File: src/app/models/coach-profile/service/coach-tests-store.service.ts (hunk 6 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/service/coach-tests-store.service.ts]: Don't need, remove b6aebad9c2c3ea2c4d560dcc67d65c872c630642

File: src/app/models/coach-profile/service/coach-tests-store.service.ts (hunk 7 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/service/coach-tests-store.service.ts]: Don't need, remove b6aebad9c2c3ea2c4d560dcc67d65c872c630642

File: src/app/models/coach-profile/service/get-coach-tests-http.service.ts (hunk 8 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/service/get-coach-tests-http.service.ts]: Change to string  b6aebad9c2c3ea2c4d560dcc67d65c872c630642

File: src/app/models/coach-profile/service/coach-tests-store.service.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      .pipe(concatMap(() => this.coachTestHttpService.getHighPriorityCoachTests()))
+      .subscribe({
+        next: (res) => {
+          this.coachTestResults = [...res];
+        },
+      });
+  }
+
+  getCoachCheckedTestResults(): void {
+    this.coachTestResults$
```
Comment (RomPavel) [inline src/app/models/coach-profile/service/coach-tests-store.service.ts:31]: Why you need this?)

Comment (ExxZzyy) [reply src/app/models/coach-profile/service/coach-tests-store.service.ts:31]: You mean getCoachCheckedTestResults function?

Comment (RomPavel) [reply src/app/models/coach-profile/service/coach-tests-store.service.ts:31]: I mean wait for this.coachTestResults$ in all functions

Comment (ExxZzyy) [reply src/app/models/coach-profile/service/coach-tests-store.service.ts]: Done 
348ac75c0b0280e6ab0768fcea8f76bd0b04f690

File: src/app/models/coach-profile/service/get-coach-tests-http.service.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { BASE_API_URL } from '../../../constants/route-constant';
+
```
Comment (shuliakleonid) [inline src/app/models/coach-profile/service/get-coach-tests-http.service.ts:6]: Interface must be in another folder, for example interfaces

**Review by shuliakleonid — CHANGES_REQUESTED** [review]

File: src/app/models/coach-profile/service/get-coach-tests-http.service.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+export interface CoachTest {
+  id: string;
+  level: string;
+  isChecked: boolean;
+  isHigh: boolean;
+  date: string;
+}
```
Comment (shuliakleonid) [inline src/app/models/coach-profile/service/get-coach-tests-http.service.ts:13]: Interface must be in another folder, for example interfaces

**Review by shuliakleonid — CHANGES_REQUESTED** [review]

File: src/app/models/coach-profile/service/get-coach-tests-http.service.ts [hunk 12]
```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { BASE_API_URL } from '../../../constants/route-constant';
+
+export interface CoachTest {
+  id: string;
+  level: string;
```
Comment (shuliakleonid) [inline src/app/models/coach-profile/service/get-coach-tests-http.service.ts:9]: Interface must be in another folder, for example interfaces

**Review by shuliakleonid — CHANGES_REQUESTED** [review]

File: src/app/models/coach-profile/service/get-coach-tests-http.service.ts (hunk 12 above)
Comment (ExxZzyy) [reply src/app/models/coach-profile/service/get-coach-tests-http.service.ts]: Done
f629f6e82a3497ccf39cbc5e62f1daf69638aaf0

**Review by shuliakleonid — APPROVED** [review]

**Pull request merged**